### ISSUE TITLE
XSLT parameters

### ISSUE THREAD

**Issue by mlcollard** [srcML/srcML#1571]
The current choices and naming of client options for passing parameters to XSLT programs is not consistent. Take the following XSLT program:

```XML
<xsl:stylesheet xmlns:src="http://www.srcML.org/srcML/src" xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">

<xsl:output method="xml" version="1.0" encoding="UTF-8" standalone="yes"/>

<!-- external parameter -->
<xsl:param name="language"/>

<!-- change language attribute -->
<xsl:template match="src:unit/@language"><xsl:attribute name="language"><xsl:value-of select="$language"/></xsl:attribute></xsl:template>

<!-- default copy -->
<xsl:template match="@*|node()"><xsl:copy><xsl:apply-templates select="@*|node()"/></xsl:copy></xsl:template>

</xsl:stylesheet>
```

with the following srcML:

```XML
<unit xmlns="http://www.srcML.org/srcML/src" xmlns:cpp="http://www.srcML.org/srcML/cpp" revision="0.9.5" language="C++">
<expr_stmt><expr><name>a</name></expr>;</expr_stmt></unit>
```

We currently have one option for passing the parameter to set the language:

```bash
srcml --xpathparam language="Java"
```

We can also apply xsltproc to it with the option ```--stringparam```

```bash
xsltproc --stringparam language Java setlanguage.xsl -
```

xsltproc has another option for passing the parameter, ```--param```

```bash
xsltproc --param language \"Java\" setlanguage.xsl -
```

Note that the string "Java" has to be wrapped with escaped newlines, i.e., the following does NOT work:

```bash
xsltproc --param language Java setlanguage.xsl -
```

To summarize for xsltproc
* ```--stringparam``` pass a string value as an argument to an xsl:param (what most people want)
*```--param``` pass a (potentially) non-string value to an xsl:param

For ```--param```, the only usage I can think of is passing a numeric value

So back to srcml client. We currently have:

* ```--xpathparam``` Maps to xsltproc ```--stringparam```

First, that is a bad, misleading name, and needs to change. Proposing the following

* ```--xslt_param``` maps to xsltproc ```--stringparam```
* ```--xslt_value_param``` maps to xsltproc ```--param```
